tests/infrastructure/instance_types/test_common_vm_preference.py

```diff
@@ -59,31 +61,45 @@
     "rhel.8.dpdk",
     "rhel.9.dpdk",
 ]
+GENERIC_FIRMWARE_PREFERENCE_LIST = [
+    "linux",
+    "linux.efi",
+    "legacy",
+]
 
 
-def start_vm_with_cluster_preference(client, preference_name, namespace_name):
-    cluster_preference = VirtualMachineClusterPreference(name=preference_name)
-    cluster_spec = cluster_preference.instance.spec
-    cpu_guest = cluster_spec.requirements.cpu.guest
-    spread_options = getattr(cluster_spec.cpu, "spreadOptions", None)
+def _extract_resources_from_cluster_spec(cluster_spec):
+    cpu_guest = cluster_spec.get("requirements", {}).get("cpu", {}).get("guest")
+    memory_guest = (
+        cluster_spec.get("requirements", {}).get("memory", {}).get("guest") or Images.Rhel.DEFAULT_MEMORY_SIZE
+    )
+    spread_options = cluster_spec.get("cpu", {}).get("spreadOptions", {})
 
-    sockets = cpu_guest
+    sockets = None
     cores = None
     threads = None
 
-    # Default ratio is 2 if not specified
-    if spread_options:
-        cores = spread_options.get("ratio", 2)
+    if cpu_guest:
```

Comment:
nitpick but consider using walrus:
```suggestion
    if cpu_guest := cluster_spec.get("requirements", {}).get("cpu", {}).get("guest"):
```